### ISSUE TITLE
Unusually long time for P2.

### ISSUE THREAD

**Issue by prodchia** [madMAx43v3r/chia-plotter#241]
I have been running madMax plotter with 1hr plot time for last 12 hours or so. In the current plot, phase took astonishing 16290 seconds to finish. Any idea what could have happened that lead to his sudden slow down?

Plot Name: plot-k32-2021-06-11-02-35-b9e1aab6edc4a1e1a77c641ae49cf534f394d4420f8c74fe647db553f2aca1bc
[P1] Table 1 took 29.9074 sec
[P1] Table 2 took 214.201 sec, found 4295005489 matches
[P1] Table 3 took 315.524 sec, found 4295057223 matches
Copy to r:\plot-k32-2021-06-11-01-29-55489713abbe23eddc30269597a20f37fceeb395e2d0a3c52312249139fb81a2.plot finished, took 593.934 sec, 0 MB/s avg.
[P1] Table 4 took 328.578 sec, found 4294810902 matches
[P1] Table 5 took 319.154 sec, found 4294811055 matches
[P1] Table 6 took 308.385 sec, found 4294694511 matches
[P1] Table 7 took 227.841 sec, found 4294452397 matches
Phase 1 took 1743.8 sec
[P2] max_table_size = 4295057223
[P2] Table 7 scan took 9.89308 sec
[P2] Table 7 rewrite took 49.2832 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 25.7577 sec
[P2] Table 6 rewrite took 36.2782 sec, dropped 581267140 entries (13.5345 %)
[P2] Table 5 scan took 26.4928 sec
[P2] Table 5 rewrite took 35.1635 sec, dropped 761992080 entries (17.7422 %)
[P2] Table 4 scan took 25.748 sec
[P2] Table 4 rewrite took 34.3136 sec, dropped 828803890 entries (19.2978 %)
[P2] Table 3 scan took 25.6453 sec
[P2] Table 3 rewrite took 33.8894 sec, dropped 855176989 entries (19.9107 %)
[P2] Table 2 scan took **15929.4 sec**
[P2] Table 2 rewrite took 35.1902 sec, dropped 865586936 entries (20.1533 %)
Phase 2 took **16290.4** sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 103.059 sec, wrote 3429418553 right entries

**Comment by madMAx43v3r** [comment]
that table is read from `tmpdir`, maybe some issue with the drive?

**Comment by grobalt** [comment]
P2 is not scaling for me as well .. i compared ramdrive for tmp2 and tmp1 a bad nvme. 4/8/16/18 threads and improvement for P2 is really small. Disk is always at 100% load in P2.

**Comment by madMAx43v3r** [comment]
yes P2 mostly reads and writes to `tmpdir`, only table 7 is done via `tmpdir2`.

**Comment by prodchia** [comment]
> that table is read from `tmpdir`, maybe some issue with the drive?

That's what I thought. I can't find any issue. I am using the "unofficial" windows version.

**Comment by grobalt** [comment]
![image](https://user-images.githubusercontent.com/79084968/121684469-049e9c80-cabf-11eb-9a78-927ef26d6c06.png)

**Comment by madMAx43v3r** [comment]
> > that table is read from `tmpdir`, maybe some issue with the drive?
> 
> That's what I thought. I can't find any issue. I am using the "unofficial" windows version.

Let's see if it happens again and how often

**Comment by hidoba** [comment]
This happened to me once on Windows, I originally thought the Windows terminal hanged and instinctively pressed Enter. Immediately after that the next line appeared (Table scan finished, enormous number of seconds).

**Comment by madMAx43v3r** [comment]
if you click on windows terminal it locks it (so you can select text), once you click again it becomes unlocked and new text will show
